jecs.luau

```diff
@@ -945,6 +975,13 @@ local function world_set(world: ecs_world_t, entity: i53, id: i53, data: unknown
 	end
 
 	if from then
+		if ECS_IS_PAIR(id) and bit32.band(idr.flags, ECS_ID_EXCLUSIVE) ~= 0 then
+			local on_remove = idr_hooks.on_remove
+			if on_remove then
+				on_remove(entity)
+			end
+		end
+
 		-- If there was a previous archetype, then the entity needs to move the archetype
 		entity_move(entity_index, entity, record, to)
```

Comment:
It has to be record.archetype since on_remove can change the archetype it is destined for.